client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -750,7 +745,7 @@ private SendResult sendKernelImpl(final Message msg,
                 }
 
                 final String tranMsg = msg.getProperty(MessageConst.PROPERTY_TRANSACTION_PREPARED);
-                if (tranMsg != null && Boolean.parseBoolean(tranMsg)) {
+                if (Boolean.parseBoolean(tranMsg)) {
```

Comment:
`tranMsg != null` can be viewed as double check here although `Boolean.parseBoolean` will check it, so it seems that removing this null check won't gain much profit.